### PR TITLE
Fixed bug APPM-1132, APPM-1131, APPM-1130, APPM-1128

### PR DESCRIPTION
**Pull request by sajithab**

### CODE REVIEW

File: modules/jaggery-apps/app-store-web/themes/store/js/handlebars.js [hunk 1]
```diff
@@ -117,6 +117,51 @@ Handlebars.registerHelper('with', function(context, options) {
 Handlebars.registerHelper('log', function(context) {
   Handlebars.log(context);
 });
+
```
Comment (ruwanta) [inline modules/jaggery-apps/app-store-web/themes/store/js/handlebars.js]: It would be good if we put an example usage in a comment

Comment (sajithab) [inline modules/jaggery-apps/app-store-web/themes/store/js/handlebars.js]: This handlebar will called by the hbs files. Putting hbs cade comment in a js file gives kind of an ambiguity meaning.

File: modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/assets.jag [hunk 2]
```diff
@@ -218,23 +220,23 @@ require('/modules/store.js').exec(function (ctx) {
...
                 };
 
                 assets = [];
                 var aid, asset, done = false;
-                while (assets.length < 12 && maxTry < 10 && !done) {
+                while (assets.length < PAGE_SIZE && maxTry < 10 && !done) {
                     maxTry++;
                     var result = getNextassetSet();
-                    if (result.length < 12){
+                    if (result.length < PAGE_SIZE){
```
Comment (darshanasbg) [inline modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/assets.jag]: Fix formatting issue.

File: modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/assets.jag [hunk 3]
```diff
@@ -255,29 +257,37 @@ require('/modules/store.js').exec(function (ctx) {
...
-    AuthService=require(getServiceDir('authentication.js')).serviceModule;
-    authenticator=new AuthService.Authenticator();
-    authenticator.init(jagg,session);
-    user=authenticator.getLoggedInUser();
+        // Query the assets with the assumed 'next page'
+        var nextAssetSet = store.assets(type, nextPage);
 
-    if(currentPage-1 > 0){
-        var leftnav = currentPage-1;
+        if(nextAssetSet.length > 0){
```
Comment (darshanasbg) [inline modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/assets.jag]: fix formatting issue.

**Pull request merged**